### PR TITLE
Rm branching from pr lesson

### LINKED ISSUE #7: Remove/Fix branching in PR lesson

### PR DESCRIPTION
**Pull request by sstevens2**
The PR lesson needed a lot of work, it was very preliminary.

Changes include:
- Adding all of the pictures into the repo
- Updating some of the pictures to use branches
- Removing the broken and unneeded branching section
- Added lots more detail to directions

I'd love lots of feedback on this.  I'm betting there is plenty of room for improvement.
One of the things I'm not sure about is I added a few callouts that point out places where you can't just copy the command you need to replace parts with the right username or organization.  Not sure if they are needed or not?  Let me know what you think.

I also didn't add in a conflict challenge yet.  I will I just thought I'd get the PR started for this so you could start to review it when you have time.

Closes #7

### CODE REVIEW

**Comment by sstevens2** [comment]
Added new section with conflict in PR for issue #6  in commit 	c4da130.

File: _episodes/09-pull-requests.md [hunk 1]
```diff
@@ -55,25 +85,38 @@ Unpacking objects: 100% (6/6), done.
...
 
-Copy the web address for this repo (from the web address line or click the 'clone and download' and copy that).
+Copy the web address for this repo 
+(from the web address line or click the 'clone and download' and copy that).
 
 ![](../fig/github_screenshot_upstream_repo.png)
 
 Then back in your terminal, navigate in the cloned repo and add the remote connection.
-For this command we must give the new connection a different nickname, where our original connection is 'origin'
+For this command we must give the new connection a different nickname,
```
Comment (sstevens2) [inline _episodes/09-pull-requests.md]: Thanks!  Good idea.  I guess I must think of them as connections but the consistency is better.

File: _episodes/09-pull-requests.md [hunk 2]
```diff
@@ -84,32 +127,66 @@ $ git remote -v
...
+upstream	https://github.com/INSTRUCTOR-GIVEN/countries.git (fetch)
+upstream	https://github.com/INSTRUCTOR-GIVEN/countries.git (push)
 ~~~
 {: .output}
 
-Now that we have this setup done, which is only needed the first time, we can now suggest changes to this repo.
-Each person will add info about a specific country to the repo.
-Since you cloned the repo a template file with info on the United States has been added.
-First we will update our local version of the repo, with this change.
+Now that we have this setup done, which is only needed the first time,
```
Comment (sstevens2) [inline _episodes/09-pull-requests.md]: Do you think it would be okay to get rid of that phrase all together? I was thinking 
"Now that we have this setup done we will be able to suggest changes to this repo using a pull request."

File: _episodes/09-pull-requests.md [hunk 3]
```diff
@@ -133,17 +210,18 @@ $ git commit -m "Added file on france"
...
 {: .output}
 
 Now we can push those changes to our forked version of the repo.
 Since we don't have permissions to the upstream repo or we want to suggest these changed be
 reviewed, if we do have permissions, we won't push directly to the upstream repo.
+Remember we need to push the name of our new branch.
 
 ~~~
-$ git push origin master
+$ git push origin addFrance
```
Comment (sstevens2) [inline _episodes/09-pull-requests.md]: Thanks I missed the change in the output block there

File: _episodes/09-pull-requests.md [hunk 4]
```diff
@@ -160,23 +238,40 @@ To https://github.com/USERNAME/countries.git
...
+to add for the person who reviews your suggestion.
+Then you can click the 'create pull request button' to submit the pull request.
 
 ![](../fig/github_screenshot_makingPR2.png)
 
 Now someone with privileges to the upstream repo can review it, give comments and
 suggestions, and merge it into the upstream version.
 
-Let's add a line to our file and see how/if this changes our pull request.
+Our collaborator reviewing the pull request noticed that
```
Comment (sstevens2) [inline _episodes/09-pull-requests.md]: I think having another instructor leave a comment is a great idea!  Not sure if it is needed to show them the other instructor's screen (unless we do inline comments).
I will try to add some sentences about what you can see in a PR.

File: _episodes/09-pull-requests.md (hunk 2 above)
Comment (pshriwise) [inline _episodes/09-pull-requests.md]: Yes! That sounds good to me.

File: _episodes/09-pull-requests.md [hunk 5]
```diff
@@ -1,51 +1,81 @@
...
-$ cd Desktop
+$ cd ~/Desktop
 $ git clone https://github.com/USERNAME/countries.git
 ~~~
 {: .bash}
 
+> ## Why does the command above say 'USERNAME'?
+>
+> So that we can't copy the command above and accidently clone someone else's
+> version of countries to our computer the command above uses the placeholder
```
Comment (pshriwise) [inline _episodes/09-pull-requests.md]: Some punctuation is needed on this line between "computer" and "the". Either a period or comma will do I think.

File: _episodes/09-pull-requests.md [hunk 6]
```diff
@@ -84,32 +128,66 @@ $ git remote -v
...
-First we will update our local version of the repo, with this change.
+Now that we have this setup done we will be able to suggest 
+changes to this repo using a pull request.
+Each person will add a new file with info about a new country in it.
+
+The instructor will now add a single file to the repository containing 
+information about the the United States.
+
+Next, we will update our local version of the repo, with this change by `pull`ing down
+the changes from the 'upstream' version's master branch.
```
Comment (pshriwise) [inline _episodes/09-pull-requests.md]: *warning: pedantic*

It may be better to use remote or repository in place of version here for consistency.

File: _episodes/09-pull-requests.md [hunk 7]
```diff
@@ -84,32 +128,66 @@ $ git remote -v
...
 ~~~
 {: .bash}
 
-Now your local version of the repo is updated but our forked version of the repo is not yet up to date.
-Now we will update the our forked version.
+Now your local version of the repo is updated but our forked version of the 
+repo is not yet up to date.
+You can reload your fork in github and see it does not contain the new 
+`united_states.txt` file.
+Now we need to update the our forked version.
```
Comment (pshriwise) [inline _episodes/09-pull-requests.md]: Extra "the" can go in this line.

File: _episodes/09-pull-requests.md [hunk 8]
```diff
@@ -84,32 +128,66 @@ $ git remote -v
...
 ~~~
 {: .bash}
 
-Now you can add your country to repo.
-First copy `united_states.txt` and change the name to the name of your chosen country.
-Then edit the contents to reflect the info of your country.
+Now let's each add a new country to the repository.
+First let's make a new branch to work on.  This will keep our 'master' version
+in sync with the authoritative version of the repository.
+We can name our branch descriptively after the country we will be adding
```
Comment (pshriwise) [inline _episodes/09-pull-requests.md]: Punctuation needed at the end of this line.

File: _episodes/09-pull-requests.md [hunk 9]
```diff
@@ -84,32 +128,66 @@ $ git remote -v
...
+Now let's each add a new country to the repository.
+First let's make a new branch to work on.  This will keep our 'master' version
+in sync with the authoritative version of the repository.
+We can name our branch descriptively after the country we will be adding 
+Mine will be `addFrance` since I'll be working with France.
+Please pick a different country and shout it out (or add it to the etherpad) 
+so no one else chooses the same one.
+We will create the branch and switch into in one step 
+as we learned earlier in the branching lesson.
+Finally before we proceed to adding the new file, we will double
```
Comment (pshriwise) [inline _episodes/09-pull-requests.md]: I'm feeling that this paragraph should be broken up and the commands separated for clarity and reduced cognitive load here. I'm thinking:

  - checkout in one step description. end paragraph
  - `git checkout -b addFrance`
  - "Finally before we proceed..."
  - `git branch`
  - output from `git branch` command

File: _episodes/09-pull-requests.md [hunk 10]
```diff
@@ -192,99 +291,224 @@ Largest City: Paris
 {: .output}
 
 Next we will add and commit these changes.
-Then we will push them to our forked version of the repo.
```
Comment (pshriwise) [inline _episodes/09-pull-requests.md]: There are quite a few places (some of them legacy) where "forked version" is used.

I think I prefer just using the term `fork` instead to get them familiar with that nomenclature. Maybe mixing this term in or providing extra description in parenthesis once or twice for clarity is a nice middle ground.

File: _episodes/09-pull-requests.md [hunk 11]
```diff
@@ -133,17 +211,18 @@ $ git commit -m "Added file on france"
...
 
 ~~~
-[master 79a312a] Added file on france
+[addFrance 79a312a] Added file on france
  1 file changed, 2 insertions(+), 2 deletions(-)
 ~~~
 {: .output}
 
 Now we can push those changes to our forked version of the repo.
 Since we don't have permissions to the upstream repo or we want to suggest these changed be
```
Comment (pshriwise) [inline _episodes/09-pull-requests.md]: This sentence was somewhat confusing to me at first read, thought I think I get what is trying to be conveyed here. 

It could also be stated: "In the case that we don't have permission to push directly to the upstream repo or that we would like our changes to be reviewed regardless, we won't push directly to the upstream repository. Instead we will push the name of our new branch to the remote linked to our fork, origin."

File: _episodes/09-pull-requests.md [hunk 12]
```diff
@@ -155,28 +234,48 @@ Writing objects: 100% (4/4), 783 bytes | 0 bytes/s, done.
 Total 4 (delta 3), reused 0 (delta 0)
 remote: Resolving deltas: 100% (3/3), completed with 3 local objects.
 To https://github.com/USERNAME/countries.git
-   2037539..79a312a  master -> master
+   2037539..79a312a  addFrance -> addFrance
 ~~~
 {: .output}
 
 Next go to your forked github version of the repo and reload the page.
```
Comment (pshriwise) [inline _episodes/09-pull-requests.md]: I didn't notice this image the first time through. I think it's super important to have them go through this step and see the new branch on GitHub. Nice!

File: _episodes/09-pull-requests.md [hunk 13]
```diff
@@ -155,28 +234,48 @@ Writing objects: 100% (4/4), 783 bytes | 0 bytes/s, done.
...
 
-![](../fig/github_screenshot_origin_wFrance.png)
+Then you should be able to view the files and commits in that branch.
+
+![](../fig/github_screenshot_origin_master_addFrance.png)
+
+Github already suspects that we are going to want to make a pull request so we can click
+the 'Compare & pull request' button to start a new pull request.
 
 Then click the 'New pull request' button.
```
Comment (pshriwise) [inline _episodes/09-pull-requests.md]: Is this line still needed?

File: _episodes/09-pull-requests.md [hunk 14]
```diff
@@ -155,28 +234,48 @@ Writing objects: 100% (4/4), 783 bytes | 0 bytes/s, done.
...
 
 ![](../fig/github_screenshot_makingPR1.png)
 
-The base fork should be the upstream version and then the head fork should be your version.
-Then click the 'create pull request button'.
+The base fork should be the upstream/authoritative version's master branch and then
```
Comment (pshriwise) [inline _episodes/09-pull-requests.md]: I hadn't noticed that they used the term `fork` here in GH. That's unfortunate IMO. Potentially confusing to learners :disappointed:

File: _episodes/09-pull-requests.md [hunk 15]
```diff
@@ -192,99 +291,224 @@ Largest City: Paris
...
+Delta compression using up to 4 threads.
+Compressing objects: 100% (3/3), done.
+Writing objects: 100% (3/3), 387 bytes | 0 bytes/s, done.
+Total 3 (delta 0), reused 0 (delta 0)
+To https://github.com/sstevens2/countries.git
+   31aa2e3..609acfe  addFrance -> addFrance
+~~~
+{: .output}
+
 Now if we reload the pull request.  The new commit was added to that pull request.
```
Comment (pshriwise) [inline _episodes/09-pull-requests.md]: (Legacy lines) My suggestion is: "If we reload the pull request, we'll see that the new commit was added to the pull request and the changes have been automatically updated."

File: _episodes/09-pull-requests.md [hunk 16]
```diff
@@ -192,99 +291,224 @@ Largest City: Paris
...
+New commits pushed to the same branch are included in the previous pull request.
+If you want to suggest changes separately you need to make separate branches but 
+if you want the changes to be considered together you should put them in the same branch.
+
+![](../fig/github_screenshot_after_new_commit.png)
+
+When working with others we might encounter the conflicts, which we
```
Comment (pshriwise) [inline _episodes/09-pull-requests.md]: Feel free to ignore this, but I think, in the context of the full set of lessons for this workshop, that this could be separated into an exercise to make it clear that it is separate from the core content of this particular lesson. What do you think?

File: _episodes/09-pull-requests.md [hunk 17]
```diff
@@ -192,99 +291,224 @@ Largest City: Paris
...
+{: .bash}
+
+~~~
+* addFrance
+  master
+~~~
+{: .output}
+
+
+Next we will each add our country to the existing `README.md` file in the repository in the line directly following the `Countries:` line.
```
Comment (pshriwise) [inline _episodes/09-pull-requests.md]: I really like the concept of creating a conflict in the README. It's a great way to introduce README's in repos and their automatic rendering on GH. (This may be a repeat comment. If so, still love it. :smile: )

File: _episodes/09-pull-requests.md [hunk 18]
```diff
@@ -192,99 +291,224 @@ Largest City: Paris
...
+Countries:
+France
+~~~
+{: .output}
+
+Next we need to add, commit, and push these requests to our existing pull request.
+
+~~~
+$ git add README.md
+$ git commit -m "Added France to list of countries in README"
```
Comment (pshriwise) [inline _episodes/09-pull-requests.md]: *warning pedantic* Should there be output for the commit command here?

File: _episodes/09-pull-requests.md [hunk 19]
```diff
@@ -192,99 +291,224 @@ Largest City: Paris
...
+{: .output}
+
+Now if we reload the page we had a pull request we notice that our `addFrance`
+branch is conflicting with upstream's `master` branch.
+This is because someone else edited the same line of the `README.md` file by 
+adding 'United States' where we added 'France'.
+
+![](../fig/github_screenshot_conflicting_PR.png)
+
+It is possible to resolve this conflict in github by clicking the 'Resolve Conflicts' button.
```
Comment (pshriwise) [inline _episodes/09-pull-requests.md]: It might be worth while to add "In this case, it is possible..."

Otherwise learners might feel we're putting them through unnecessary pains in handling the conflict manually.

File: _episodes/09-pull-requests.md [hunk 20]
```diff
@@ -192,99 +291,224 @@ Largest City: Paris
...
+From https://github.com/McMahonLab/countries
+ * branch            master     -> FETCH_HEAD
+Auto-merging README.md
+CONFLICT (content): Merge conflict in README.md
+Automatic merge failed; fix conflicts and then commit the result.
+~~~
+{: .output}
+
+
+From the conflict error message we can see the conflict is in `README.md` or by running `git status` and seeing the 'both modified' status.
```
Comment (pshriwise) [inline _episodes/09-pull-requests.md]: I like the multiple ways of figuring out what's going on with the conflict. Directing learners back to `git status` when in doubt is always good.

File: _episodes/09-pull-requests.md (hunk 6 above)
Comment (sstevens2) [inline _episodes/09-pull-requests.md]: In new version I changed version to remote and fleshed out the description a bit.  Let me know if you think it needs more work (when I push next).

P.S. I really appreciate your noticing when it isn't simplified enough.  Very helpful!

File: _episodes/09-pull-requests.md (hunk 11 above)
Comment (sstevens2) [inline _episodes/09-pull-requests.md]: I made some changes here (will get pushed but copied below).  I edited your suggestion a little but I still feel like I missing the motivation a little here.  I want to get across that the pull request is useful for suggesting changes, esp if you don't have push permissions or want someone to review.  I might try working more on the language here.  Let me know if you have suggestions.

"Now we can push those changes to our forked version of the repo.
In some cases we may not have permission to push directly to the 
upstream repo or we might like our changes to be reviewed regardless 
of permissions, so we won't push directly to the upstream repository. 
Instead we will push the name of our new branch to the remote linked 
to our fork, `origin`."

File: _episodes/09-pull-requests.md (hunk 12 above)
Comment (sstevens2) [inline _episodes/09-pull-requests.md]: Might be an added image... I added in a few.

File: _episodes/09-pull-requests.md (hunk 13 above)
Comment (sstevens2) [inline _episodes/09-pull-requests.md]: Nope, don't think so.  Good catch!

File: _episodes/09-pull-requests.md (hunk 14 above)
Comment (sstevens2) [inline _episodes/09-pull-requests.md]: Yeah, I think the wording for these two branches when making a PR is really bad.  It confuses even seasoned git users which is the head and which is the base.

File: _episodes/09-pull-requests.md (hunk 10 above)
Comment (sstevens2) [inline _episodes/09-pull-requests.md]: I'll remove 'version'.  I think this is a hold over from how I think about repos but it is probably not a good term to use for learners.

Comment (sstevens2) [inline _episodes/09-pull-requests.md]: found another place where I removed version and replaced with fork as well

File: _episodes/09-pull-requests.md (hunk 11 above)
Comment (pshriwise) [inline _episodes/09-pull-requests.md]: I think it would help to remove the first line. Why we're pushing to the fork should probably be be explained. This assertion that we're going to do so appears abruptly. I like the way you modified the second sentence here. I think the first half of that sentence lays out the two main motivations for using PR's to suggest changes. The reason that we're pushing to the fork is sort of a by product of setting up a PR but it isn't framed that way. Right now there's a gap between identifying a situation in which you want to make a PR and pushing to the leaner's fork. 

I'll propose another iteration:

"In some cases we may not have permission to push changes directly to the upstream repo or we might like our changes to be reviewed regardless of permissions, so we'll create a `pull request`. A `pull request` is a **request** for a member of the upstream repository to **pull** our changes into the upstream repository from a `fork`, allowing them to request further changes/improvement and make comments on the changes before doing so. In order to create a `pull request`, we must push our new branch containing the changes we'd like to submit to the remote linked to our fork, `origin`, on GitHub."

Comment (pshriwise) [inline _episodes/09-pull-requests.md]: edit: upstream repo could arguably be replaced with authoritative repo?

File: _episodes/09-pull-requests.md (hunk 16 above)
Comment (sstevens2) [inline _episodes/09-pull-requests.md]: Yeah I've been conflicted about this as well.  I was thinking about adding this as an exercise but it is something you have to do with the instructor (since it requires two people making the same changes).  I kind of see it more as an optional instructor lead exercise?  So if you don't get to it it is okay but if you have time, I think it would be fun.

I think I'm going to leave it for now but we could discuss more about if we should cut it out or rearrange it or something after the workshop.

File: _episodes/09-pull-requests.md (hunk 18 above)
Comment (sstevens2) [inline _episodes/09-pull-requests.md]: Yeah this was probably me being lazy.  Added in most recent version.

File: _episodes/09-pull-requests.md (hunk 19 above)
Comment (sstevens2) [inline _episodes/09-pull-requests.md]: I had never done it in github so I actually tested out resolving it in github and I liked it better locally but this might just be my preference.  I was considering adding that info in but think it might just be a personal opinion.

File: _episodes/09-pull-requests.md (hunk 20 above)
Comment (sstevens2) [inline _episodes/09-pull-requests.md]: Agreed, think that was your suggestion, so thanks!  I ❤️ `git status`, I always *sort of* joke about that it is my fav command when I teach git 😉 because I'm always running it to remember where I am in the process.

**Comment by sstevens2** [comment]
Hey @pshriwise, Thanks for all the feedback.  Seems like it is getting better.  Commit 3d3c7b2
 should have the changes based on your most recent feedback in it.  Including typos, rewording and adding some output where there was none before.   Let me know if you see anything else!

File: _episodes/09-pull-requests.md (hunk 11 above)
Comment (sstevens2) [inline _episodes/09-pull-requests.md]: Beautiful!  Like your proposed change a lot.  Will edit that and make another commit now.

**Comment by sstevens2** [comment]
Commit 26f9dd4 adds in @pshriwise wording suggested for introducing PRs

**Comment by sstevens2** [comment]
I was thinking once we merge this, we might send this lesson to the swc-dc list if people want to look it over and make issues for anything they notice.  What do you think?  This should wait until you are happy with it and ready to merge it though, then we can send them the updated rendered website page.

File: _episodes/09-pull-requests.md (hunk 20 above)
Comment (pshriwise) [inline _episodes/09-pull-requests.md]: :flushed: #selflike

File: _episodes/09-pull-requests.md (hunk 19 above)
Comment (pshriwise) [inline _episodes/09-pull-requests.md]: I agree. Also I think for the purpose of understanding and enabling the learners to handle tricky situations like this it is better do teach them how to do it locally.

**Comment by pshriwise** [comment]
@sstevens2 I think that is a great idea :+1:

File: _episodes/09-pull-requests.md [hunk 21]
```diff
@@ -192,99 +305,237 @@ Largest City: Paris
...
+
+> ## Add new country file and make additional PR
 >
-> - Copy your first country file into a new country
+> - Starting in the master branch make a new branch
+> - Copy other country file into a new country
 > - Edit the file to include info on the new country
 > - Add and commit this new file
 > - Push the new changes to github
 > > ## Solution
```
Comment (pshriwise) [inline _episodes/09-pull-requests.md]: This exercise isn't rendering :cry: 

I *think* there is an empty line (only a `>`) missing before the solution section.

Comment (sstevens2) [inline _episodes/09-pull-requests.md]: Added line.  Hope it solved this issue?

Comment (pshriwise) [inline _episodes/09-pull-requests.md]: Looks good to me!

**Comment by pshriwise** [comment]
Everything looks good to me! Thank you again @sstevens2!
